src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/CustomDatasetSourceFactory.kt

```diff
@@ -0,0 +1,8 @@
+package com.atlassian.performance.tools.awsinfrastructure.api
+
+interface CustomDatasetSourceFactory {
```

Comment:
If the factory has to exist, then let's move it next into `CustomDatasetSource`, just like its `Builder`.

Reply:
I will remove the factory as you suggested in the other comment, however just hypothetically I'm not sure how this move should be implemented. Would the factory still be an interface? Would it be `CustomDatasetSource.Factory`? I guess the interface method could evolve together with the constructor of `CustomDatasetSource.Builder`.

Reply:
> Would it be `CustomDatasetSource.Factory`?

Yes. Same type, different location.